flintrock/templates/spark/conf/spark-env.sh

```diff
@@ -6,14 +6,21 @@ export SPARK_LOCAL_DIRS="{root_ephemeral_dirs}"
 export SPARK_EXECUTOR_INSTANCES="1"
 export SPARK_WORKER_CORES="$(nproc)"
 
-export SPARK_MASTER_HOST="{master_host}"
+export SPARK_MASTER_HOST="{master_ip}"
+
+# Needed for spark 1.6.x
+export SPARK_MASTER_IP="{master_ip}"
```

Comment:
Yes, because we use both spark 1.6 and spark 2 : 
- in spark 2, scripts start-master.sh and start-slaves.sh use SPARK_MASTER_HOST
- in spark 1.6, scripts start-master.sh and start-slaves.sh use SPARK_MASTER_IP

See this pull request for more details: https://github.com/apache/spark/pull/13543/files